pkg/controller/sparkapplication/controller.go

```diff
@@ -523,17 +540,30 @@ func (c *Controller) updateSparkApplicationStatusWithRetries(
 	}
 
 	if lastUpdateErr != nil {
-		glog.Errorf("failed to update SparkApplication %s: %v", toUpdate.Name, lastUpdateErr)
+		glog.Errorf("failed to update SparkApplication %s: %v", original.Name, lastUpdateErr)
 	}
 
 	return nil
 }
 
+func (c *Controller) postMetrics(oldApp, newApp *v1alpha1.SparkApplication) {
+
+	if c.metrics == nil {
```

Comment:
Like mentioned above, these two conditions can be moved out of this function.

Reply:
removed this function